packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -73,20 +72,17 @@ export function streamFromString(str: string): ReadableStream<Uint8Array> {
 export async function streamToString(
   stream: ReadableStream<Uint8Array>
 ): Promise<string> {
-  let buffer = ''
-
-  await stream
-    // Decode the streamed chunks to turn them into strings.
-    .pipeThrough(createDecodeTransformStream())
-    .pipeTo(
-      new WritableStream<string>({
-        write(chunk) {
-          buffer += chunk
-        },
-      })
-    )
+  const decoder = new TextDecoder()
```

Comment:
should probably use `{ fatal: true }` option since we control the inputs and there is nothing we would expect to be undecodable. Probably best to raise this as an error eagerly